src/main/java/com/cloudbees/plugins/credentials/CredentialsProvider.java

```diff
@@ -1510,20 +1510,20 @@ public static Fingerprint getOrCreateFingerprintOf(@NonNull Credentials c) throw
     public static <C extends Credentials> List<C> trackAll(@NonNull Node node, @NonNull List<C> credentials) {
         long timestamp = System.currentTimeMillis();
         String nodeName = node.getNodeName();
+        // Create a list of all current node names. The credential will only be
+        // fingerprinted if it is one of these.
+        // TODO (JENKINS-51694): This breaks tracking for cloud agents. We should
+        // track those agents against the cloud instead of the node itself.
+        Set<String> jenkinsNodeNames = new HashSet<>();
+        // TODO: Switch to Jenkins.get() once 2.98+ is the baseline
+        for (Node n: Jenkins.getActiveInstance().getNodes()) {
+            jenkinsNodeNames.add(n.getNodeName());
+        }
```

Comment:
I moved the set creation out of the `for (Credentials c : credentials)` loop so we don't have to recreate it every time through.